src/runtime-rs/crates/resource/src/share_fs/share_virtio_fs.rs

```diff
@@ -112,19 +119,28 @@ pub async fn rafs_mount(
         sl!(),
         "Attaching rafs meta file {} to virtio-fs device, rafs mount point {}", rafs_meta, rafs_mnt
     );
-    let virtio_fs = ShareFsMountDevice {
-        config: ShareFsMountConfig {
-            source: rafs_meta.clone(),
-            fstype: ShareFsMountType::RAFS,
-            mount_point: rafs_mnt,
-            config: Some(config_content),
-            tag: String::from(MOUNT_GUEST_TAG),
-            op: ShareFsOperation::Mount,
-            prefetch_list_path,
-        },
+
+    let rafs_config = ShareFsMountConfig {
+        source: rafs_meta.clone(),
+        fstype: ShareFsMountType::RAFS,
+        mount_point: rafs_mnt,
+        config: Some(config_content),
+        tag: String::from(MOUNT_GUEST_TAG),
+        op: ShareFsOperation::Mount,
+        prefetch_list_path,
+    };
+
+    let shared_path = utils::get_host_ro_shared_path(sid).display().to_string();
+    let sharefs_config = ShareFsConfig {
+        shared_path,
+        mount_config: Some(rafs_config),
+        ..Default::default()
     };
-    h.add_device(DeviceType::ShareFsMount(virtio_fs))
+
+    // update virtio-fs device with ShareFsMountConfig
+    do_update_device(d, &DeviceConfig::ShareFsCfg(sharefs_config.clone()))
```

Comment:
Just like a similar piece of code in `setup_inline_virtiofs()`, is `clone()` necessary here?

Reply:
Fixed